plugins/metrics/pkg/agent/node.go

```diff
@@ -149,13 +171,66 @@ func (m *MetricsNode) GetHealth(_ context.Context, _ *emptypb.Empty) (*corev1.He
 	}, nil
 }
 
+// Start Implements remoteread.RemoteReadServer
+
+func (m *MetricsNode) Start(ctx context.Context, request *remoteread.StartReadRequest) (*emptypb.Empty, error) {
+	m.targetRunnerMu.Lock()
+	defer m.targetRunnerMu.Unlock()
+
+	if err := m.targetRunner.Start(request.Target, request.Query); err != nil {
+		return nil, err
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (m *MetricsNode) Stop(ctx context.Context, request *remoteread.StopReadRequest) (*emptypb.Empty, error) {
+	m.targetRunnerMu.Lock()
+	defer m.targetRunnerMu.Unlock()
+
+	if err := m.targetRunner.Stop(request.Meta.Name); err != nil {
+		return nil, err
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (m *MetricsNode) GetTargetStatus(ctx context.Context, request *remoteread.TargetStatusRequest) (*remoteread.TargetStatus, error) {
+	m.targetRunnerMu.RLock()
+	defer m.targetRunnerMu.RUnlock()
+
+	return m.targetRunner.GetStatus(request.Meta.Name)
+}
+
+func (m *MetricsNode) Discover(ctx context.Context, request *remoteread.DiscoveryRequest) (*remoteread.DiscoveryResponse, error) {
+	driver, err := drivers.GetNodeDriver("external-operator")
```

Comment:
You should add the `Discover` method to the driver interface, feel free to refactor as necessary (it'll probably look weird but in this case it's okay for the driver interface to be a collection of unrelated methods). Maybe also rename to `Discover{RemoteReadTargets|Targets|Prometheuses|etc}`.

Adding the method will also require an implementation in the testenv driver, which will definitely be useful for integration tests.